Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModulePresenter.java

```diff
@@ -1,40 +1,19 @@
 package com.microsoft.tooling.msservices.serviceexplorer.azure.webapp;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
 
-import com.microsoft.azure.management.appservice.WebApp;
-import com.microsoft.azuretools.core.mvp.model.ResourceEx;
 import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
 import com.microsoft.azuretools.core.mvp.ui.base.MvpPresenter;
 
-public class WebAppModulePresenter<V extends WebAppModule> extends MvpPresenter<WebAppModule> {
+public class WebAppModulePresenter<V extends WebAppModuleView> extends MvpPresenter<V> {
     /**
      * Called from view when the view needs refresh.
      */
     public void onModuleRefresh() {
-        List<ResourceEx<WebApp>> webapps = new ArrayList<>();
-        webapps.addAll(AzureWebAppMvpModel.getInstance().listAllWebApps(true));
-
-        if (getMvpView() == null) {
-            return;
+        final WebAppModuleView view = getMvpView();
+        if (view != null) {
+            view.renderChildren(AzureWebAppMvpModel.getInstance().listAllWebApps(true));
```

Comment:
` AzureWebAppMvpModel.getInstance().listAllWebApps(true)` should be executed in another thread, in order not to block the UI thread.